api/src/main/java/org/apache/iceberg/io/InputFile.java

```diff
@@ -36,6 +36,14 @@ public interface InputFile {
    */
   long getLength();
 
+  /**
+   * Returns the modification time of the file, in milliseconds
+   *
+   * @return the modification time of the file, in milliseconds
+   * @throws RuntimeIOException If the implementation throws an {@link IOException}
+   */
+  long getModifiedTime();
```

Comment:
same comment as above about naming and potentially making this a `default` method so that we don't introduce API-breaking changes